lib/cocoapods/installer.rb

```diff
@@ -445,6 +449,18 @@ def run_plugins_post_install_hooks
       HooksManager.run(:post_install, context, plugins)
     end
 
+    # Runs the registered callbacks for the source provider plugin hooks.
+    #
+    # @param [Analyzer] analyzer The analyzer that will be used to process sources
+    #
+    # @return [void]
+    #
+    def run_source_provider_hooks
+      context = SourceProviderHooksContext.generate(podfile)
+      HooksManager.run(:source_provider, context, plugins)
+      context.sources
```

Comment:
can we think of a way to do this without allowing other plugins to access each other's sources? That feels a bit weird to me -- maybe the context could have an `add_source` method?